woocommerce/woocommerce-ios
# Adding bulk price settings view controller

Part of: #6242 & #6239

### Description
Adds the view controller that allows the user to enter a new regular price and bulk update all the product variations.
This view controller is pushed when the regular price cell is tapped in the bulk update screen #6239.
The view controller has a tableview with a single cell `UnitInputTableViewCell` that contains the textfield where the user enters the price. The tableView does not extent to cover the whole view, an area at the bottom of the view is the place of the save button(of type `ButtonActivityIndicator`). The keyboard is monitored using the `KeyboardFrameObserver` and the constraint of the save button to the bottom is adjusted in order to always be 16 points above the keyboard.

The `UnitInputTableViewCell` currently has the same style as in the single variation price update, I will provide the style to match the mockups in a followup pull request(that has only the price input on the left, we have a sub-task for this).
Price change events are send to the view model, and it updates the state of the button accordingly. The view controller receives events for the state of the button and is enabled if we have a non empty price.

When the save button is tapped the API call is triggered, the keyboard is dismissed and for the duration of the call the button displays an activity indicator(and can not be tapped).
If the bulk update is successful the view controller is popped from the navigation stack. In case of error the user will remain in the current screen. In both cases an `Notice` will be displayed that will be added in a follow up pull request(, we have a sub-task for this).  
The view controller also has a footer text that displays a text informing the user about the current price of the variations and the number of variations to be updated. Plural variations have been taken in to account. To create this text we are passing in the initializer of the `BulkUpdatePriceSettingsViewModel` a  BulkUpdateOptionsModel instead of an array of product variations.

### Testing instructions
1. Goto the products tab
2. Select a product with variations
3. Go to the list of variations
4. Tap on the "more" button in the navigation bar
5. Select the "Bulk update"
6. The bulk update screen should appear, select the regular price option
7. The price setting screen should appear
8. Enter a valid price, the save button should be enabled only if the price is non empty
9. Bellow the price field a text should be displayed that depending if the variations have the same price or not the footer text have the following values:
Current price is <formated price>. The price will be updated for <number of variations> variation[s].
Current prices are mixed.. The price will be updated for <number of variations> variation[s].
Current price is not set. The price will be updated for <number of variations> variation[s].
10. Tap save button, the keyboard should be dismissed and a loading indicator should be displayed in the save button
11. The price should be updated and the user returned to the previous bulk update screen




https://user-images.githubusercontent.com/96764631/157611395-15e18cf0-defc-4108-8e0c-b7724a86af07.mp4



https://user-images.githubusercontent.com/96764631/157609296-a1ccfea3-cd5f-4820-9d39-c53adf6650bb.mp4



---
